### PR TITLE
Cdc mutation properties

### PR DESCRIPTION
**Pull request by rneher**
this PR makes changes to enable processing of NA segments along with HA. It also calculates a number of extra features (incl age distribution and glycosylation count) that can be visualized in auspice.

### CODE REVIEW

**Comment by trvrb** [comment]
@rneher: I'm worried about this code here: https://github.com/nextstrain/augur/blob/cdc_mutation_properties/builds/flu/run_flu.py#L70. As I understand it, `flu.prepare` uses `ensure_all_segments` (https://github.com/nextstrain/augur/blob/cdc_mutation_properties/builds/flu/flu.prepare.py#L139). This will cause us to remove HA segments that don't have a matching NA. Looking at GISAID now for viruses _collected_ after Dec 1, I see 91 viruses, of which 75 have HA+NA and 91 just have HA. I would recommend to to just do independent segment analyses for the moment. Especially while we don't have segments tied together. I don't want to lose recent HA viruses.

**Comment by trvrb** [comment]
This commit (26dbe4925c3096a4d15d2d549b21a4da6d00831a) is my suggested revision.

**Comment by rneher** [comment]
The `ensure_all_segments`  is useful when looking at trees side by side. but I see your concern. I think I would prefer to handle this in `prepare` with an additional flag. I'll look at it tonight.

**Comment by rneher** [comment]
how about this? `ensure_all_segments` is now set as an argument to `prepare`. It was a flag in `config` all along.

File: builds/flu/flu_titers.py [hunk 1]
```diff
@@ -56,7 +56,122 @@ def HI_export(process):
...
+        else:
+            node.attr["num_gender"] = 0
+
+    for node in tree.get_nonterminals(order='postorder'):
+        node.tip_count = np.sum([c.tip_count for c in node])
+        node.attr['num_gender'] = np.sum([c.attr['num_gender']*c.tip_count for c in node])/node.tip_count
+
+
+    # not sure those have any effects right now... but auspice 2.0 might use them
+    runner.config["auspice"]["color_options"]["glyc"] = {
```
Comment (trvrb) [inline builds/flu/flu_titers.py]: @rneher: Shouldn't this be in `flu.process.py` and not in `flu_titers.py` along with other `runner.config` options?

Comment (trvrb) [reply builds/flu/flu_titers.py]: I see... this fits with placement of `ep`, `ne` and `rb`. Seems generally a little weird to have `H3N2_scores` as part of `flu_titers` to me.

**Comment by trvrb** [comment]
This now builds fully and produces the same JSON structure as current `master`, ie: `flu_h3n2_ha_3y_tree.json`. Nothing has to change in auspice to load these. It also produces `flu_h3n2_na_3y_tree.json`, but this requires a second call to `run_flu.py` with the `--process_na` flag.

Currently not implemented are auspice changes to add additional coloring and changes to include index files for NA. I would propose to merge this now, so we have a working build on `master` and then do a separate PR for auspice changes.

@rneher: Thank you so much for getting this stuff implemented. I think they're really great directions.

**Pull request merged**